Sources/Hummingbird/Utils/DateCache.swift

```diff
@@ -12,61 +12,61 @@
-/// update a cached version of the date in the format as detailed in RFC1123 once every second. To
-/// avoid threading issues it is assumed that `currentDate` will only every be accessed on the same
-/// EventLoop that the update is running.
-public final class HBDateCache {
-    /// Current formatted date
-    public var currentDate: String
+/// Getting the current date formatted is an expensive operation. This creates a task that will
+/// update a cached version of the date in the format as detailed in RFC1123 once every second.
+public final class HBDateCache: Service {
+    final class DateContainer: AtomicReference, Sendable {
+        let date: String
 
-    /// return date cache for this thread. If one doesn't exist create one scheduled on EventLoop
-    public static func getDateCache(on eventLoop: EventLoop) -> HBDateCache {
-        guard let dateCache = thread.currentValue else {
-            self.thread.currentValue = .init(eventLoop: eventLoop)
-            return self.thread.currentValue!
+        init(date: String) {
+            self.date = date
         }
-        return dateCache
     }
 
-    static func shutdownDateCaches(eventLoopGroup: EventLoopGroup) -> EventLoopFuture<Void> {
-        var dateCacheShutdownFutures: [EventLoopFuture<Void>] = []
-        for eventLoop in eventLoopGroup.makeIterator() {
-            let future: EventLoopFuture<Void> = eventLoop.flatSubmit {
-                guard let dateCache = thread.currentValue else {
-                    return eventLoop.makeSucceededFuture(())
-                }
-                self.thread.currentValue = nil
-                return dateCache.shutdown(eventLoop: eventLoop)
-            }
-            dateCacheShutdownFutures.append(future)
-        }
-        return EventLoopFuture.andAllComplete(dateCacheShutdownFutures, on: eventLoopGroup.next())
-    }
+    let dateContainer: ManagedAtomic<DateContainer>
 
-    /// Initialize DateCache to run on a specific `EventLoop`
-    private init(eventLoop: EventLoop) {
-        assert(eventLoop.inEventLoop)
-        var timeVal = timeval.init()
-        gettimeofday(&timeVal, nil)
-        self.currentDate = Self.formatRFC1123Date(timeVal.tv_sec)
+    init() {
+        let epochTime = time(nil)
+        self.dateContainer = .init(.init(date: Self.formatRFC1123Date(epochTime)))
+    }
 
-        let millisecondsSinceLastSecond = Double(timeVal.tv_usec) / 1000.0
-        let millisecondsUntilNextSecond = Int64(1000.0 - millisecondsSinceLastSecond)
-        self.task = eventLoop.scheduleRepeatedTask(initialDelay: .milliseconds(millisecondsUntilNextSecond), delay: .seconds(1)) { _ in
-            self.updateDate()
+    public func run() async throws {
+        let cancelled = ManagedAtomic(false)
+        if #available(macOS 13.0, *) {
+            let timerSequence = AsyncTimerSequence(interval: .seconds(1), clock: .continuous)
```

Comment:
This can be a `.suspending` clock. We don't hold a second counter ourselves, but request system time. So I think we should suspend.